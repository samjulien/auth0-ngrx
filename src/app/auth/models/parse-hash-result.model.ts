export interface ParseHashResult {
  error: any;
  authResult: any;
}
