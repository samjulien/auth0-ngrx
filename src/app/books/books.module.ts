import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { ComponentsModule } from './components';
import { MaterialModule } from '../material';

import { reducers } from './reducers';
import { BooksPageEffects } from './effects/books-page.effects';
import { BooksPageComponent } from './components/books-page.component';
import { AuthGuardService } from '@app/auth/services/auth-guard.service';

@NgModule({
  imports: [
    CommonModule,
    MaterialModule,
    ComponentsModule,
    RouterModule.forChild([
      { path: '', component: BooksPageComponent, canActivate: [AuthGuardService] },
    ]),
    StoreModule.forFeature('books', reducers),
    EffectsModule.forFeature([BooksPageEffects]),
  ]
})
export class BooksModule {}
