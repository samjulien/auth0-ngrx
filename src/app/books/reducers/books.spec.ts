import * as fromBooks from './books';
import { Load } from '../actions/books-page.actions';
import { Book, generateMockBook } from '../models/book';
import { LoadSuccess } from '../actions/books-api.actions';

describe('Books Reducer', () => {
  const book1 = generateMockBook();
  const book2 = { ...book1, id: '222' };
  const initialState: fromBooks.State = {
    ids: [book1.id, book2.id],
    entities: {
      [book1.id]: book1,
      [book2.id]: book2,
    }
  };

  describe('State Changes', () => {
    it('should have an initial state', () => {
      const state = fromBooks.reducer(initialState, { type: '@@init' } as any);

      expect(state).toBe(initialState);
    });

    it('should load books on success', () => {
      const init = { type: '@@init' } as any;
      const load = new LoadSuccess([ book1, book2 ]);

      const state = [init, load].reduce(fromBooks.reducer, initialState);

      expect(state).toMatchSnapshot();
    });    
  });

  describe('Selectors', () => {
    describe('getAllBooks', () => {
      it('should return all the books', () => {
        const result = fromBooks.getAllBooks(initialState);

        expect(result.length).toBe(2);
      });
    });
  });
});
