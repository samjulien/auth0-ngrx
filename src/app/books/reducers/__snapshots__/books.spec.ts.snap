// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books Reducer State Changes should load books on success 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "id": "1",
      "volumeInfo": Object {
        "authors": Array [
          "author",
        ],
        "averageRating": 3,
        "description": "description",
        "imageLinks": Object {
          "smallThumbnail": "string",
          "thumbnail": "string",
        },
        "publishDate": "",
        "publisher": "publisher",
        "ratingsCount": 5,
        "subtitle": "subtitle",
        "title": "title",
      },
    },
    "222": Object {
      "id": "222",
      "volumeInfo": Object {
        "authors": Array [
          "author",
        ],
        "averageRating": 3,
        "description": "description",
        "imageLinks": Object {
          "smallThumbnail": "string",
          "thumbnail": "string",
        },
        "publishDate": "",
        "publisher": "publisher",
        "ratingsCount": 5,
        "subtitle": "subtitle",
        "title": "title",
      },
    },
  },
  "ids": Array [
    "1",
    "222",
  ],
}
`;
